### PR TITLE
Allow a group to require 2fa

### LINKED ISSUE #133: Allow a group to enforce 2fa
Access should be denied to this group's permissions, and this group should not be included in a user's groups, until 2fa is enabled when required

### PR DESCRIPTION
**Pull request by ChrisMacNaughton**
Closes #133

### CODE REVIEW

File: app/controllers/basic_auth_controller.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class BasicAuthController < ApplicationController
   def create
     if user_signed_in?
       permission_checks = [params[:permission_name], "#{params[:permission_name]}.#{params[:format]}"]
-      groups = current_user.groups
+      groups = current_user.asserted_groups
```
Comment (igalic) [inline app/controllers/basic_auth_controller.rb:9]: what's the difference between these two?

**Review by igalic — COMMENTED** [review]
two questions, one inline, and the other: you gonna add tests here?

**Comment by ChrisMacNaughton** [comment]
This change does [add tests](https://github.com/CentauriSolutions/EyeDP/pull/144/files#diff-67cc913763e1d78ac097b2a82cf5296c070d548c1b706396dd9035d7f7382e14R50) as well!

File: app/controllers/basic_auth_controller.rb (hunk 1 above)
Comment (ChrisMacNaughton) [reply app/controllers/basic_auth_controller.rb:9]: Groups is the base relation between users and groups (when an admin adds a user to a group, they'd show up here), but asserted_groups is a [new function in this change](https://github.com/CentauriSolutions/EyeDP/pull/144/files#diff-9802ca3c9c4cf89904fd44bc114e35ebdf2c5dd3d5b645491e2b253e1afef29bR135-R142) that checks for 2fa enabled, and then potentially returns a subset of groups the user is a member of.

**Review by igalic — APPROVED** [review]
:+1:

**Pull request merged**